asana/asana.go

```diff
@@ -189,8 +183,39 @@ type (
 
 	// Errors always has at least 1 element when returned.
 	Errors []Error
+
+	RequestError struct {
+		Err      error
+		Response *http.Response
+		Text     string
+	}
 )
 
+func (re RequestError) New(resp *http.Response, err error) RequestError {
```

Comment:
Idiomatically I think this would normally just be `func NewRequestError`. But I also don't think it needs to be exported since we don't expect anything out of the SDK to create errors. Fine to just inline or have as an internal method.